fs/span-manager/span_manager.go

```diff
@@ -162,6 +163,49 @@ func (m *SpanManager) buildAllSpans() {
 	}
 }
 
+// FetchSpanBackground invokes the reader to fetch the span in the background and cache it.
+// It is invoked by the backgroundFetcher.
+func (m *SpanManager) FetchSpanInBackground(spanID soci.SpanID) error {
+	if spanID > m.ztoc.CompressionInfo.MaxSpanID {
+		return ErrExceedMaxSpan
+	}
+
+	s := m.spans[spanID]
+	s.mu.Lock()
+	state := s.state.Load().(spanState)
+
+	// Only fetch if the span hasn't been requested yet.
+	if state != unrequested {
+		s.mu.Unlock()
+		return ErrSpanStateNotUnrequested
+	}
+
+	s.setState(requested)
+	s.mu.Unlock()
+
+	compressedSize := s.endCompOffset - s.startCompOffset
+	compressedBuf := make([]byte, compressedSize)
+
+	n, err := m.r.ReadAt(compressedBuf, int64(s.startCompOffset))
+	if err != nil {
+		return err
+	}
+
+	if n != len(compressedBuf) {
+		return fmt.Errorf("unexpected data size for reading compressed span. read = %d, expected = %d", n, len(compressedBuf))
+	}
+
+	if err := m.verifySpanContents(compressedBuf, spanID); err != nil {
+		return err
+	}
+
+	m.addSpanToCache(fmt.Sprintf("%d", spanID), compressedBuf)
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	return s.setState(fetched)
+}
+
 func (m *SpanManager) ResolveSpan(spanID soci.SpanID, r *io.SectionReader) error {
```

Comment:
This function can be removed.

Reply:
There are a bunch of tests that rely on that method. I don't want to remove it just yet, will remove in a future PR.